pkg/core/blockchain.go

```diff
@@ -3018,7 +3018,7 @@ func (bc *Blockchain) verifyTxWitnesses(t *transaction.Transaction, block *block
 // verifyHeaderWitnesses is a block-specific implementation of VerifyWitnesses logic.
 func (bc *Blockchain) verifyHeaderWitnesses(currHeader, prevHeader *block.Header) error {
 	var hash util.Uint160
-	if prevHeader == nil && currHeader.PrevHash.Equals(util.Uint256{}) {
+	if currHeader.PrevHash.Equals(util.Uint256{}) {
 		hash = currHeader.Script.ScriptHash()
```

Comment:
It seems like a dead code to me. @roman-khimov, do we really need this path? Also, verifying against the block's own Script's hash makes almost no sense.

Reply:
Genesis block never faces this code anyway.

Reply:
This case can be valid only for genesis. Genesis block never runs this code. Throw it away at last, please.